src/comms/config_layer.cpp

```diff
@@ -1,21 +1,80 @@
 #include "config_layer.hpp"
 
 const Config* const ConfigLayer::configure(HIDLayer* comms) {
+    // attempt SD card load
+    config_SD_init(comms);
+    if (configured) return &config;
+
+    // if no config on SD, then await transmission
     // grab and process all config packets until finished
+    uint32_t prev_time = millis();
+    uint32_t delta_time = 0;
     while (!is_configured()) {
+    #ifdef CONFIG_LAYER_DEBUG
+        if (delta_time >= 2000) {
+            Serial.printf("Pinging for config packet....\n");
+            prev_time = millis();
+        }
+        delta_time = millis() - prev_time;
+    #endif
         comms->ping();
         process(comms->get_incoming_packet(), comms->get_outgoing_packet());
     }
 
     // put the data from the packets into the config object
     config.fill_data(config_packets, subsec_sizes);
 
+    // verify that config received matches ref system: if not, error out
+#ifndef CONFIG_OFF_ROBOT
+    Serial.printf("Received robot ID from config: %d\nRobot ID from ref system: %d\n", (int)config.robot_id, ref.ref_data.robot_performance.robot_ID);
+    if ((ref.ref_data.robot_performance.robot_ID % 100) != (int)config.robot_id) {
+        Serial.printf("ERROR: IDs do not match!! Check robot_id.cfg and robot settings from ref system!\n");
+        if (!CONFIG_ERR_HANDLER(CONFIG_ID_MISMATCH)) {
+            // in current implementation, CONFIG_ERR_HANDLER w/ err code CONFIG_ID_MISMATCH will
+            // enter an infinite while(1) loop-- if that is changed, this loop should be changed accordingly as well
+            while (1) {
+                Serial.println("CONFIG_ERR_HANDLER: exited with error code CONFIG_ID_MISMATCH");
+                Serial.println("if function was modified for case CONFIG_ID_MISMATCH, remove this loop in config_layer.cpp");
+                delay(2000);
+            }
+        }
+    }
+#endif
+
+// update stored config, msg if successful
+    Serial.println("Attempting to store config...");
+    if (store_config()) Serial.println("Config successfully stored in /config.pack");
+    else Serial.println("Config not successfully stored (is an SD card inserted?)");    // not a fatal error, can still return
+
     return &config;
 }
 
-void ConfigLayer::process(CommsPacket *in, CommsPacket *out) {
-    char *in_raw = in->raw;
-    char *out_raw = out->raw;
+void ConfigLayer::config_SD_init(HIDLayer* comms) {
+    const char* config_path = "/config.pack";
```

Comment:
It would probably be better to store this in the header so you dont need to declare it everywhere you need it.

Reply:
By this, do you mean to simply declare a macro for it (`#include CONFIG_PATH "/config.pack"`)?

Reply:
Yeah, that or something like `const char* CONFIG_PATH = "blah.pack"`. I just want it to be easily changeable.

Reply:
Looks good though